site/source/tutorials/1010_containers_and_injecting_state/index.md

```diff
@@ -0,0 +1,177 @@
+
+{% task 'Create a class to manage application state.' %}
+
+To begin our discussion, let's take a look at the initial version of `App`:
+
+{% include_codefile 'demo/initial/biz-e-corp/src/widgets/App.ts' %}
+
+Most of this widget is dedicated to holding and managing the `WorkerData` in the application. Notice, however, that it never actually uses that data itself. `App` is only holding the data so that its children can access it. While this may be acceptable in a small application, in larger applications, the `App` class can become overwhelmed by the amount of state information that it is tracking. Since the state information not a primary concern of the `App` class, let's refactor it out of `App` and into a new class.
+
+{% instruction 'Add the following to `ApplicationContext.ts`.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/ApplicationContext.ts' %}
+
+The code begins by importing some modules, including the `WorkerProperties` and `WorkerFormData` interfaces defined in the `Worker` and `WorkerForm` modules. These two interfaces define the state that the `ApplicationContext` is going to track. One important item to note is that the `ApplicationContext` is aware of the widgets, but the widgets are not aware of the `ApplicationContext` module. It is important to import the modules this way since it minimizes the widgets' dependencies and makes them easier to test.
+
+After the imports, we have defined the `defaultWorkerForm`. This is identical to the `defautlForm` from the `App` module and is used for the same purpose - to reset the form data when a new `Worker` is created.
+
+The `ApplicationContext` class contains the state information. It contains two private fields, `_workerData` and `_formData` that will contain the state as well as two accessor methods to retrieve those fields. Finally, notice the two public methods that `ApplicationContext` exposes:
+
+{% include_codefile 'demo/finished/biz-e-corp/src/ApplicationContext.ts' lines:32-41 %}
+
+The `formInput` method is providing the same functionality that the `_onFormInput` method in the `App` class and the `submitForm` method is doing the same thing that the `App` class's `_addWorker` is. The implementations are slightly different since the `ApplicationContext` has dedicated fields that are storing the state information. Also, since the `ApplicationContext` is not a widget, it needs to emit an `invalidate` event to inform the application to schedule a render cycle.
+
+Notice that the `ApplicationContext` does not have any code to load state information. Its role is only to manage the application's state. Any data that the application should start with must be provided to the `ApplicationContext` via its `constructor` while the application is being started.
+
+Now that we have moved state management to a dedicated module, we need a way to initialize the state and make it available. We will do that in the next section where we will create and register an `Injector`.
```

Comment:
"We will do that in the next section where we will create and register an `Injector`" -> "We will initialize state by creating and registering an `Injector`"